### PR TITLE
Implement feature #7: Calculate individual expenses

### LINKED ISSUE #7: Issue 7 — Per person summary (`Reports.perPerson`)
**Goal:** For each person compute:
- `paidTotal`: sum of expenses they paid
- `owedTotal`: sum of shares they owe as participant
- `net = paidTotal − owedTotal`

**Acceptance Criteria:**
- Returns `SimplePersonSummaryMap` keyed by name.
- Consistent with balances model.

### PR DESCRIPTION
**Pull request by BirkLauritzen**
This is a pull request with a contribution that fixes issue 7 and tests for it as well.

This PR solves https://github.com/ST-ITM-2025-2/expense-splitter/issues/7

Implemented the CSV file writing logic in the ExpenseStore append method.
Implemented the expenseToCSV method to convert an Expense object into a CSV string.
Added the ReportTest file
How to test: Run the program -> input 6 and enter -> the amount will now be divided between the participants

happy_path:
Correctly dividing the amount with the participants

edge_case:
empty participant list
empty expense amount
Closes https://github.com/ST-ITM-2025-2/expense-splitter/issues/7

### CODE REVIEW

**Comment by hayat711** [comment]
**Group8** will be reviewing this PR [code implemented by group 9 for feature #7 ]

File: src/main/java/itm/oss/splitter/Reports.java [hunk 1]
```diff
@@ -1,16 +1,74 @@
...
+            ArrayList<String> sortedParticipants = new ArrayList<>(participants);
+            Collections.sort(sortedParticipants);
+
+            for (int i = 0; i < sortedParticipants.size(); i++) {
+                String name = sortedParticipants.get(i);
+
+                if (map.get(name) == null) {
+                    map.put(name, new PersonSummary(BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO));
+                }
+                PersonSummary ps = map.get(name);
```
Comment (saschahuberzh) [inline src/main/java/itm/oss/splitter/Reports.java]: Consider renaming ps to something more descriptive like personSummary, it would make the loop logic easier to follow

File: src/main/java/itm/oss/splitter/ExpenseStore.java [hunk 2]
```diff
@@ -1,15 +1,52 @@
...
-    // TODO (Issue 1): parse CSV file into Expense list.
-    // Format: date,payer,amount,currency,participants,category,notes
-    // participants are semicolon-separated.
-    throw new UnsupportedOperationException("load() not implemented yet");
+    public ArrayList<Expense> load(String path) throws IOException {
+        // TODO (Issue 1): parse CSV file into Expense list.
+        // Format: date,payer,amount,currency,participants,category,notes
+        // participants are semicolon-separated.
+        String line = "";
+        ArrayList<Expense> expenses = new ArrayList<>();
```
Comment (saschahuberzh) [inline src/main/java/itm/oss/splitter/ExpenseStore.java]: Using the List interface instead of ArrayList in the method signature would make the code more flexible and easier to maintain in the future

File: src/main/java/itm/oss/splitter/Reports.java [hunk 3]
```diff
@@ -1,16 +1,74 @@
...
+                BigDecimal owed = baseShare;
+                if (i < remainder.multiply(new BigDecimal("100")).intValue()) {
+                    owed = owed.add(new BigDecimal("0.01"));
+                }
+
+                ps.addOwed(owed);
+            }
+        }
+
+        for (String name : map.keys()) {
```
Comment (MHJeong730) [inline src/main/java/itm/oss/splitter/Reports.java:88]: I can't figure out what this for loop is doing. It seems to simply add 0 to owedTotal. Is there a specific function intended for this for loop? Could you please add some comments on what this loop does?

File: src/main/java/itm/oss/splitter/Reports.java [hunk 4]
```diff
@@ -1,16 +1,74 @@
...
+                if (map.get(name) == null) {
+                    map.put(name, new PersonSummary(BigDecimal.ZERO, BigDecimal.ZERO, BigDecimal.ZERO));
+                }
+                PersonSummary ps = map.get(name);
+
+                if (name.equals(expense.getPayer())) {
+                    ps.addPaid(amount);
+                }
+
+                // credit to Kasper (github: kasperB2004) for helping to fix the cent rounding problem
```
Comment (MHJeong730) [inline src/main/java/itm/oss/splitter/Reports.java]: I think this idea of ​​solving the problem of sharing indivisible expenses is excellent.

File: src/test/java/itm/oss/splitter/ReportsTest.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        SimplePersonSummaryMap map = reports.perPerson(xs);
+
+        // Sanity on loader
+        assertEquals(3, xs.size(), "Should load three expense rows (header skipped)");
+        assertEquals("Alice", xs.get(0).getPayer(), "the payer should be alice");
+        assertEquals(new BigDecimal("60.00"), xs.get(0).getAmount(), "the amount should be 60");
+        assertEquals(3, xs.get(0).getParticipants().size(), "There should be three participants");
+    }
+
+// Unit testing
```
Comment (MHJeong730) [inline src/test/java/itm/oss/splitter/ReportsTest.java:66]: Adding test cases for various edge cases is very helpful. However, it would be better if there were test cases for the getNet() function to ensure that the recalculateNet() function works properly.

File: src/main/java/itm/oss/splitter/Reports.java (hunk 3 above)
Comment (Driedoutjerky) [reply src/main/java/itm/oss/splitter/Reports.java:88]: The for-loop code you've mentioned adds `BigDecimal.ZERO` to the `owedTotal` for each name in the `personSummary`.
That ensures `recalculateNet()` method to run for the final time for each participant before returning `map` to ensure that no stale numbers have been left un-calculated. @Poppaluu will be adding the comment to the code soon. Thank you for your feedback!

**Comment by jobregon1212** [comment]
Hello guys, this PR should only modify the .java files (implementation and test files). It should not touch .gitignore , README, or any other file. So please fix that. Remember to synchronize with upstream repository and pull the latest changes from the upstream repository.

File: src/main/java/itm/oss/splitter/Reports.java (hunk 1 above)
Comment (Poppaluu) [reply src/main/java/itm/oss/splitter/Reports.java]: ps renamed to personSummary

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 2 above)
Comment (Poppaluu) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: ArrayLists now use the List interface

File: src/main/java/itm/oss/splitter/Reports.java (hunk 4 above)
Comment (Poppaluu) [reply src/main/java/itm/oss/splitter/Reports.java]: ty

File: src/main/java/itm/oss/splitter/Reports.java (hunk 3 above)
Comment (Poppaluu) [reply src/main/java/itm/oss/splitter/Reports.java:88]: comments added

File: src/test/java/itm/oss/splitter/ReportsTest.java (hunk 5 above)
Comment (BirkLauritzen) [reply src/test/java/itm/oss/splitter/ReportsTest.java:66]: I've added two more edge cases, and test cases for getNet as well

**Comment by BirkLauritzen** [comment]
All the reviewed changes have been implemented and resolved

File: src/test/java/itm/oss/splitter/ReportsTest.java [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+
+// complex csv file test
+
+
+    // Issue 1 integration testing
+    @Test
+    void perPerson_usesCsvLoadedByExpenseStore() throws IOException, URISyntaxException {
+
+        // Load the CSV. Prefer classpath (src/test/resources), fallback to repo file (./data/expenses.sample.csv)
```
Comment (sengA7) [inline src/test/java/itm/oss/splitter/ReportsTest.java]: Hello, it's Agnes sorry i do the review part too late
but here the code to load expenses.sample.csv is duplicated two times with (perPerson_usesCsvLoadedByExpenseStore and perPerson_netFromSimpleCsv)
The current test coverage is excellent

File: src/test/java/itm/oss/splitter/ReportsTest.java [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+            assertMoney("-45.00", m.get("Bob").getNet(),   "Bob net should be -45");
+            assertMoney("-45.00", m.get("Cara").getNet(),  "Cara net should be -45");
+
+            // Invariant: sum of net == 0
+            BigDecimal sum = m.get("Alice").getNet()
+                .add(m.get("Bob").getNet())
+                .add(m.get("Cara").getNet());
+            assertEquals(0, sum.compareTo(BigDecimal.ZERO), "Sum of net must be zero");
+        }
+}
```
Comment (sengA7) [inline src/test/java/itm/oss/splitter/ReportsTest.java:277]: I have nothing to tell more, your code is very clean and organized (easy to read and undersand). Good job guys !!

File: src/test/java/itm/oss/splitter/ReportsTest.java (hunk 6 above)
Comment (BirkLauritzen) [reply src/test/java/itm/oss/splitter/ReportsTest.java]: Thank you for spotting it. The issue has been resolved now

File: src/test/java/itm/oss/splitter/ReportsTest.java (hunk 7 above)
Comment (BirkLauritzen) [reply src/test/java/itm/oss/splitter/ReportsTest.java:277]: Thanks!

**Review by MHJeong730 — APPROVED** [review]

**Comment by BirkLauritzen** [comment]
All files have been successfully merged with the files from main
